pkg/services/collector.go

```diff
@@ -0,0 +1,512 @@
+		chartsInstances, err := chartListContainer.Path(name).Children()
+		if err != nil {
+			log.Fatal(err)
+			continue
+		}
+
+		for _, chartInstance := range chartsInstances {
+			chartInstanceMap, err := chartInstance.ChildrenMap()
+			if err != nil {
+				log.Fatal(err)
+				continue
+			}
+
+			chartInstanceVersion := chartInstanceMap["version"].Data().(string)
+			if !slices.Contains(c.ChartVersionsRegistry[name], chartInstanceVersion) {
+				c.ChartVersionsRegistry[name] = append(c.ChartVersionsRegistry[name], chartInstanceVersion)
+			}
+		}
+	}
+}
+
+func (c *Collector) RefreshChartLastRevisionRegistry(secrets *v1.SecretList) {
+	for _, secret := range secrets.Items {
+		if !strings.HasPrefix(secret.Name, "sh.helm.release.v1") {
+			continue
+		}
+		secretChartName, secretRevisionInt, err := getChartNameAndRevisionFromSecretName(secret.Name)
+		if err != nil {
+			log.Fatal(err)
+			continue
+		}
+
+		c.ChartLastRevisionRegistry[secretChartName] = secretRevisionInt
+	}
+}
+
+func (c *Collector) DecodeReleaseString(release string) (*gabs.Container, error) {
+	encodedRelease, err := base64.StdEncoding.DecodeString(release)
+	if err != nil {
+		return nil, err
+	}
+	decodedReleaseReader, err := gzip.NewReader(bytes.NewReader(encodedRelease))
+	if err != nil {
+		return nil, err
+	}
+	decodedReleaseBytes, err := ioutil.ReadAll(decodedReleaseReader)
+	if err != nil {
+		return nil, err
+	}
+
+	return gabs.ParseJSON(decodedReleaseBytes)
+}
+
+func (c *Collector) PackageHelmChart(chartName string, chartVersion string, chartPath string) error {
+	var packageFiles []string
+
+	err := filepath.Walk(chartPath,
+		func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			pathIsDirectory, err := isDirectory(path)
+			if err != nil {
+				return err
+			}
+			if !pathIsDirectory {
+				packageFiles = append(packageFiles, path)
+			}
+
+			return nil
+		})
+	if err != nil {
+		return err
+	}
+
+	out, err := os.Create(fmt.Sprintf("%s-%s.tgz", chartName, chartVersion))
+	if err != nil {
+		return err
+	}
+	defer out.Close()
+
+	// Create the archive and write the output to the "out" Writer
+	err = createArchive(packageFiles, fmt.Sprintf("%s/%s/", filepath.Dir(chartPath), filepath.Base(chartPath)), fmt.Sprintf("%s/", chartName), out)
+	return err
+}
+
+func (c *Collector) AddHelmRepositories(releaseContainer *gabs.Container) error {
+	if !releaseContainer.ExistsP("chart.metadata.dependencies") {
+		return nil
+	}
+	dependencies, err := releaseContainer.Path("chart.metadata.dependencies").Children()
+	if err != nil {
+		return err
+	}
+
+	var repositoryNames []string
+
+	for _, dependency := range dependencies {
+		dependencyMap, err := dependency.ChildrenMap()
+		if err != nil {
+			return err
+		}
+
+		repositoryName := dependencyMap["name"].Data().(string)
+		repositoryUrl := dependencyMap["repository"].Data().(string)
+
+		_, err = execCommand(fmt.Sprintf("helm repo add %s %s", repositoryName, repositoryUrl))
```

Comment:
mb better to use something like this? https://github.com/mittwald/go-helm-client

Reply:
I decided to use this: https://github.com/helm/helm